mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -156,4 +202,58 @@ def SYCLCallOp : SYCL_Op<"call", []> {
   }];
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// range.get OPERATION
+////////////////////////////////////////////////////////////////////////////////
+
+def SYCLRangeGetOp
+    : SYCLMethodOpInterfaceImpl<"get", "RangeType",
+                                ["get", "operator[]"], [NoSideEffect]> {
+  let summary = "Call to range::get/operator[]";
+  let description = [{
+    This operation represents a call to the range::get/operator[] function.
+  }];
+
+  let arguments = (ins RangeMemRef:$Acc,
+                       I32:$Index,
+                       TypeAttr:$BaseType,
+                       FlatSymbolRefAttr:$BaseTypeName,
+                       FlatSymbolRefAttr:$Function,
+                       FlatSymbolRefAttr:$MangledName);
+
+  let results = (outs SYCLGetResult:$Res);
+
+  let hasVerifier = 1;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// accessor.getelementptr OPERATION
+////////////////////////////////////////////////////////////////////////////////
+
+def SYCLGEPIndex : AnyTypeOf<[IndexType, SYCL_IDType]>;
+
+def SYCLGEPOp
```

Comment:
Calling this a `GEP` operation is going to cause confusion because one think immediately of the LLVM GetElementPtr operation semantics. Can we use a different name (e.g. SYCLAccessorSubscriptOp) ?

Reply:
I think the semantics are pretty similar... Maybe we can use `SYCLAccessorGEP` instead?